routing/routing_integration_tests/speed_camera_notification.cpp

```diff
@@ -0,0 +1,161 @@
+#include "testing/testing.hpp"
+
+#include "routing/routing_integration_tests/routing_test_tools.hpp"
+
+#include "routing/route.hpp"
+#include "routing/routing_callbacks.hpp"
+#include "routing/routing_helpers.hpp"
+#include "routing/routing_session.hpp"
+#include "routing/routing_tests/tools.hpp"
+
+#include "platform/location.hpp"
+#include "platform/measurement_utils.hpp"
+
+#include <algorithm>
+#include <memory>
+
+using namespace routing;
+using namespace routing::turns;
+using namespace std;
+
+namespace
+{
+string const kCameraOnTheWay = "Speed camera on the way";
+
+location::GpsInfo MoveTo(ms::LatLon const & coords, double speed = -1)
+{
+  location::GpsInfo info;
+  info.m_horizontalAccuracy = 0.01;
+  info.m_verticalAccuracy = 0.01;
+  info.m_latitude = coords.lat;
+  info.m_longitude = coords.lon;
+  info.m_speedMpS = speed;
+  return info;
+}
+
+void ChangePosition(RoutingSession & routingSession, ms::LatLon const & coords, double speedKmPH = -1)
+{
+  static double constexpr kCoordEpsilon = 1e-6;
+
+  routingSession.OnLocationPositionChanged(MoveTo({coords.lat, coords.lon}, KMPH2MPS(speedKmPH)));
+  routingSession.OnLocationPositionChanged(MoveTo({coords.lat + kCoordEpsilon,
+                                                   coords.lon + kCoordEpsilon}, KMPH2MPS(speedKmPH)));
+}
+
+void InitRoutingSession(ms::LatLon const & from, ms::LatLon const & to, RoutingSession & routingSession)
+{
+  TRouteResult const routeResult =
+    integration::CalculateRoute(integration::GetVehicleComponents<VehicleType::Car>(),
+                                MercatorBounds::FromLatLon(from), {0., 0.},
```

Comment:
inlline comment to 0, 0
can we use Zero() here?

Reply:
done